### DISCUSSION TITLE
Auxkernel or Postprocessor to change TimeStepper

### DISCUSSION THREAD

**Discussion by fervaldies** [idaholab/moose#23414] [Q&A General] [answered]
Hello, I would like to change the dt value back to the minimum when a certain auxkernel changes the x position in the function. I have the following auxkernel:
```
[AuxKernels]
  [./add_material_in_the_ellipsoid]
    type = FunctionPathEllipsoidAux
    variable = level_set_var
    rx = 0.0018
    ry = 0.0015
    rz = 0.0015
    init_x_coords = '0.010 0.190 0.010'
    init_y_coords = '0.04175 0.04725 0.05275'
    init_z_coords = '0.016 0.016 0.016'
    temperature_pp = interpass_max_temp
    single_scan_length = '0.015 0.180 0.180'
    threshold_temperature = 320
    velocity = '0.007 -0.007 0.007'
  [../]
[]

and the following TimeStepper:

  [./TimeStepper]
    type = IterationAdaptiveDT
    growth_factor = 30
    cutback_factor = 0.03
    linear_iteration_ratio = 45
    optimal_iterations = 3 #12
    iteration_window = 1
    dt = 0.3
  [../]
```
so at some point the dt increases to 9 and I would like to reduce it to 0.3 when the 'init_x_coords' in the auxkernel changes from 0.010 to 0.190. Is there any way of doing this?
My problem is that when the dt is 9 and I want it to change to 0.3, it does one step more with dt=9 than desired and the whole simulation is wrong, as it needs to come back to 0.3 just the step before it changes to 0.190 the init_x_coords.

Another possibility that I was thinking is to change the dt when a certain postprocessor reaches a threshold temperature. Is this possible? I have a postprocessor that calculates the maximum temperature in the simulation and maybe I could use it to change the TimeStepper when it reaches 320K let's say. This is the postprocessor:

  [./interpass_max_temp]
    type = NodalExtremeValue
    variable = temp
    value_type = max
  [../]

Thank you in advance for your help

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll want to use the timestep_limiting_postprocessor option
https://mooseframework.inl.gov/source/timesteppers/IterationAdaptiveDT.html

The postprocessor should be a class derived from ElementPostprocessor and it examines the value of the aux variable everywhere to determine its own value. 
The value of the postprocessor is then the limiting time step

OR

You use the NodalExtremeValue to get the max of the field, and a ParsedPostprocessor to have some logic like:
`if (interpass_max_temp > 0.019, 0.3, 100000)`
in the function parameter

Guillaume

Guillaume

**Reply by fervaldies** [reply]
Thank you for your reply! I do not really understand the timestep_limiting_postprocessor option. Is it possible to tell to the TimeStepper that it has to be 0.3 when the max temperature in the simulation is less than 400K with this option?

**Reply by GiudGiud** [reply]
yes. 
Use a ParsedPostprocessor as the limiting postprocessor and for the function to parse:
```
if (interpass_max_temp <400, 0.3, 1e6) # 1e6 means no limit
```

**Reply by fervaldies** [reply]
Perfect, got it, thank you so much! :) Now it is working correctly
